File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/1406-subtract-the-product-and-sum-of-digits-of-an-integer.cpp
class Solution {
public:
    int subtractProductAndSum(int n) {
        int a = n;
        double product = 1;
        double sum = 0 ;
        int i;
        while (a!=0){
            i = a % 10;
            product = product * i;
            sum  = sum + i;
            a = a / 10;

        }
        int result = product - sum;
        return int(result);
        
    }
};

File: 1406-subtract-the-product-and-sum-of-digits-of-an-integer/1406-subtract-the-product-and-sum-of-digits-of-an-integer.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        a = n
        prod = 1
        sum = 0

        while (a!=0):
            i = a % 10
            
            prod = prod * i
            sum = sum + i
            a = a // 10
        res = prod - sum
        return int(res)
        
        